darts/models/forecasting/forecasting_model.py

```diff
@@ -382,7 +385,21 @@ def historical_forecasts(
 
         # iterate and forecast
         for pred_time in iterator:
-            train = series.drop_after(pred_time)  # build the training series
+            # build the training series
+            train = series.drop_after(pred_time)
+            if train_length and not isinstance(train_length, int):
+                raise_log(
+                    TypeError("If not None, train_length needs to be an integer."),
+                    logger,
+                )
+            elif (train_length is not None) and train_length < 1:
+                raise_log(
+                    ValueError("If not None, train_length needs to be positive."),
+                    logger,
+                )
+            if train_length and len(train) > train_length:
+                # take last train_length samples (drop_before?)
+                train = train[:train_length]
```

Comment:
Unless I'm missing something, that will take the _first_ `train_length` samples. I think you should write:
```suggestion
                train = train[-train_length:]
```